UraniumVisualizer/UraniumVisualizer/FileParser.cs

```diff
@@ -19,7 +36,33 @@ public FileParser(string fileName)
         /// <returns></returns>
         public IEnumerable<FunctionRecord> Parse()
         {
-            throw new NotImplementedException();
+            var stack = new Stack<ProfilerEvent>();
+
+            using (var f = new StreamReader(FileName))
+            {
+                while (!f.EndOfStream)
+                {
+                    var str = f.ReadLine();
+                    //PE, pe - ProfilerEvent
+                    var timeStampPE = int.Parse(Regex.Match(str, @"\d+").Value);
+                    var typePE = GetEventType(str[0]);
+                    var pe = new ProfilerEvent(timeStampPE, typePE);
+                    var functionName = Regex.Match(str.Substring(1), @"\D+").Value;
+
+                    if (stack.Count == 0 || pe.Type == EventType.Start)
+                    {
+                        yield return new FunctionRecord(functionName, stack.Count,
+                                                        pe.TimeStamp/*, double.PositiveInfinity*/);
+                        stack.Push(pe);
+                    }
+                    else
+                    {
+                        //double duration = pe.TimeStamp - stack.Peek().TimeStamp;
```

Comment:
Используй везде `var`:
```suggestion
                        //var duration = pe.TimeStamp - stack.Peek().TimeStamp;
```
И создай само свойство в `FunctionRecord`, зачем писать это в комментариях?